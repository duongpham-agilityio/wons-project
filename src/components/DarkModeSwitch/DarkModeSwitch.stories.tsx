import { Meta, StoryObj } from '@storybook/react';
import DarkModeSwitch from '.';

export default {
  title: 'Components/DarkModeSwitch',
  component: DarkModeSwitch,
} as Meta;

type Story = StoryObj<typeof DarkModeSwitch>;

// Dark mode switch type default
export const DarkModeSwitchDefault: Story = {
  args: {},
};
