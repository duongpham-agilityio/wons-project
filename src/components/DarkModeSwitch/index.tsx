import { Switch } from '@nextui-org/react';
import { MoonIcon } from '@/icons';

const DarkModeSwitch = () => {
  return (
    <div className="flex items-center justify-between my-11">
      <div className="flex items-center">
        <MoonIcon width="26px" height="26px" className="mr-6" />
        <label className="text-lg font-semibold">Dark mode</label>
      </div>
      <Switch
        defaultSelected
        color="success"
        size="md"
        aria-label="Automatic updates"
      />
    </div>
  );
};

export default DarkModeSwitch;
