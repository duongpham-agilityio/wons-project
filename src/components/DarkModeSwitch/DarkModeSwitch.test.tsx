import { render } from '@testing-library/react';
import DarkModeSwitch from '.';

describe('Dark Mode Switch component', () => {
  test('renders with default type', () => {
    const { container } = render(<DarkModeSwitch />);
    expect(container).toMatchSnapshot();
  });
});
