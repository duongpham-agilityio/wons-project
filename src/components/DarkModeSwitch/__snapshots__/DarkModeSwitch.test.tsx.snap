// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dark Mode Switch component renders with default type 1`] = `
<div>
  <label
    aria-label="Automatic updates"
    class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer touch-none tap-highlight-transparent select-none"
    data-selected="true"
  >
    <span
      style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
    >
      <input
        aria-labelledby=":r0:"
        checked=""
        role="switch"
        type="checkbox"
        value=""
      />
    </span>
    <span
      aria-hidden="true"
      class="px-1 relative inline-flex items-center justify-start flex-shrink-0 overflow-hidden bg-default-200 rounded-full outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[selected=true]:bg-success group-data-[selected=true]:text-success-foreground w-12 h-7 mr-2 rtl:ml-2 rtl:mr-[unset] transition-background"
    >
      <span
        class="z-10 flex items-center justify-center bg-white shadow-small rounded-full origin-right w-5 h-5 text-small group-data-[selected=true]:ml-5 rtl:group-data-[selected=true]:ml-0 rtl:group-data-[selected=true]:mr-5 transition-all group-data-[pressed=true]:w-6 group-data-[selected]:group-data-[pressed]:ml-4"
      />
    </span>
  </label>
</div>
`;
